enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,641 @@
+
+**Integration:** Integration with the product is strengthened, meaning that adding new features becomes more straightforward. Employing tightly coupled methods could lead to poor practices.
+
+**Alerting:** The controller can notify users through various channels:
+
+- HC status.conditions
+- Log entries
+- Events
+- Alerting
+- Metrics
+
+**Validations:** In this scenario, we incorporate an additional layer of validations (CEL), managed by api-machinery. This layer promptly detects any discrepancies with the API consensus and triggers early failure. Furthermore, additional validations can be employed within the controller.
+
+**Reusability:** Leveraging the existing ETCD backup API developed by the ETCD team, we will integrate the entire API into a new structure within our ETCD API field. Within the same field, we will incorporate the necessary functionality to upload binaries to remote storage such as S3 or Azure.
+
+#### Cons
+
+**Supportability:** Development of the API and support of it during the product life-cycle.
+**Complexity:** The Hosted Control Planes team will take the responsibility to stay tuned over the API changes of the ETCD Backup.
+
+
+## Proposal
+
+The proposal commences with the acknowledgement that a user requires an ETCD backup.
+
+The primary requirement is that the ETCD `managementType` must be configured as `managed`.
+
+This backup operation will be specified within the `hostedcluster.spec.etcd.managed` API custom field under a new field named `backup`. This new field will encompass two childs, with the primary one designated for the backup process itself, directly imported from the original ETCD Backup API. Comprehensive details regarding this API can be found in the official ETCD proposal and its corresponding implementation documentation.
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field is called `storeEndpoint` which basically will take care of the export of the backup object to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
+
+- **BackupPrefix:** Where the user could set some kind of prefix to identify that backup in a more readable way. This Prefix will be parsed, validated and used in the final name which will contain the **timestamp**, the **HostedCluster name** and the provided **BackupPrefix**.
+- **Credentials:** This is a `SecretReference` located in the HostedCluster namespace using the same name as the backup plus the "credentials" suffix.
+- **ServiceConfigType:** This is a enum field which will reference the remote cloud providers to upload the backup object (e.g s3, azure, odf).
```

Comment:
If we enforce that exactly one of the provider-specific configurations is set, I don't think we need another field to identify which provider is being used.

Reply:
That remarks the data type with an enum, the other is the spec. Sample:

- Data Type: https://github.com/openshift/hypershift/blob/182a25c6e3bee420cfcfbfda3dc53bd00394f8e0/api/hypershift/v1beta1/hostedcluster_types.go#L1691
- Spec: https://github.com/openshift/hypershift/blob/182a25c6e3bee420cfcfbfda3dc53bd00394f8e0/api/hypershift/v1beta1/hostedcluster_types.go#L1727-L1730